src/components/Input/Input.js

```diff
@@ -1,166 +1,53 @@
-    outline: none;
-  }
-`;
-const ErrorBoxWrapper = styled.div`
-  display: flex;
-  flex-direction: column;
-  flex-grow: 2;
-`;
-const FieldInputWrapper = styled.div`
-  display: flex;
-  position: relative;
-  &.text--input-top {
-    flex-direction: column;
-    justify-content: left;
-  }
-  &.text--input-left {
-    flex-direction: row;
-  }
-  &.text--input-disabled {
-    pointer-events: none;
-    opacity: 0.35;
-  }
-`;
 
-const FieldInputText = styled.label`
-  font-size: 12px;
-  height: 13px;
-  line-height: 1.25;
-  text-align: left;
-  color: ${colors.onyx.muted};
-  position: relative;
-  .text--input-disabled & {
-    opacity: 0.35;
-  }
-  .text--input-top & {
-    margin-bottom: 6px;
-  }
-  .text--input-left & {
-    margin-right: 11px;
-    margin-top: 10px;
-  }
-  &.text__error-top {
-    color: ${colors.alert.base};
-    margin-top: 1px;
-    font-size: 12px;
-    line-height: 1.25;
-  }
-  &.text__error-left {
-    color: ${colors.alert.base};
-    margin-top: 1px;
-  }
-`;
-
-const FieldErrorText = styled.label`
-  font-size: 12px;
-  height: 13px;
-  line-height: 1.25;
-  text-align: left;
-  color: ${colors.onyx.muted};
-  position: relative;
-  .text--input-disabled & {
-    opacity: 0.35;
-  }
-  .text--input-left & {
-    margin-right: 11px;
-    margin-top: 10px;
-  }
-  &.text__error-top {
-    color: ${colors.alert.base};
-    margin-top: 1px;
-    font-size: 12px;
-    line-height: 1.25;
-  }
-  &.text__error-left {
-    color: ${colors.alert.base};
-    margin-top: 1px;
-  }
-`;
+import sluggify from "../../utils/text";
+import INPUT_SIZES from "./constants";
+import {
+  FieldInputWrapper,
+  FieldInputBox,
+  ErrorBoxWrapper,
+  FieldInputText,
+  FieldErrorText
+} from "./Input.styles";
 
 const Input = ({
   labelPosition,
   placeholder,
   disabled,
-  width,
   label,
   name,
   errorMessage,
+  size,
   ...rest
 }) => (
   <FieldInputWrapper
     className={classNames({
       "text--input-top": labelPosition === "top",
       "text--input-left": labelPosition === "left",
       "text--input-disabled": disabled,
-      "text__error-top": labelPosition === "top" && errorMessage,
-      "text__error-left": labelPosition === "left" && errorMessage
+      text__error: errorMessage,
+      "text--input-small": size === "small",
+      "text--input-regular": size === "regular",
+      "text--input-large": size === "large"
     })}
   >
-    <FieldInputText id={sluggify(`${name + label}label`)}>
+    <FieldInputText id={sluggify(`${name + label}__label`)}>
       {label}
     </FieldInputText>
     <ErrorBoxWrapper>
       <FieldInputBox
         placeholder={disabled ? "" : placeholder}
         name={name}
-        id={sluggify(`${name + label}input`)}
+        id={sluggify(`${name + label}__input`)}
         disabled={disabled}
         {...rest}
-        aria-labelledby={sluggify(`${name + label}label`)}
+        aria-labelledby={sluggify(`${name + label}__label`)}
```

Comment:
Maybe just assign the result of invoking `sluggify` on `name + label` to a variable outside of the return statement and then concatenate it with label/input prefixed by dashes?

E.g.
```
const slug = sluggify(name + label);
return <FieldInputBox id={${slug}--input} {...props} />
```

Reply:
Fixed. Thanks.